From 9f505cde0fe33dd48a7a39dc853f90bec6ce11ca Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 25 Aug 2014 07:51:30 -0400
Subject: [tools] perf: Add support for PERF_RECORD_MISC_GUEST_USER in thread__find_addr_map()

Message-id: <1408953210-26343-203-git-send-email-jolsa@redhat.com>
Patchwork-id: 88886
O-Subject: [PATCH RHEL7.1 BZ1133083 202/322] perf tools: Add support for PERF_RECORD_MISC_GUEST_USER in thread__find_addr_map().
Bugzilla: 1133083
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1133083
https://bugzilla.redhat.com/show_bug.cgi?id=1133083

upstream
========
commit fb50bb43d051d12aad9e2337d0d5d14d0d3dd281
Author: Dongsheng Yang <yangds.fnst@cn.fujitsu.com>
Date: Fri Dec 20 15:52:56 2013 -0500

description
===========
This patch remove a TODO in thread__find_addr_map() and add support of
PERF_RECORD_MISC_GUEST_USER.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/event.c b/tools/perf/util/event.c
index fe20227..484e994 100644
--- a/tools/perf/util/event.c
+++ b/tools/perf/util/event.c
@@ -659,15 +659,10 @@ void thread__find_addr_map(struct thread *thread,
   al->level = 'g';
   mg = &machine->kmaps;
   load_map = true;
+ } else if (cpumode == PERF_RECORD_MISC_GUEST_USER && perf_guest) {
+  al->level = 'u';
  } else {
-  /*
-   * 'u' means guest os user space.
-   * TODO: We don't support guest user space. Might support late.
-   */
-  if (cpumode == PERF_RECORD_MISC_GUEST_USER && perf_guest)
-   al->level = 'u';
-  else
-   al->level = 'H';
+  al->level = 'H';
   al->map = NULL;
 
   if ((cpumode == PERF_RECORD_MISC_GUEST_USER ||
-- 
1.7.1